Fix templates with non-ascii characters.

three-comrades opened the pull request:
Fixes #1198 for me.

Two questions:
- Is it okay to just assume that files on Linux are encoded as 'utf-8', or should we guess the encoding (either by reading the file, or by following the hints given by environment variables)?
- The template string is passed to [parse_template](https://github.com/pazz/alot/blob/d03cd3f582818651157aa9588e5723cdaa81dc3e/alot/db/envelope.py#L281) which seems to require byte strings, such that headers like "Subject: =?iso-8859-1?q?p=F6stal?=" can be decoded later. But I don't think anyone would use a template file like that, such that it should be okay to decode the file when it's opened.

varac commented:
@three-comrades  wow, great - I mean the quick pull req, I can't review of judge code quality here.

dcbaker commented on alot/commands/globals.py:
```diff
@@ -770,7 +771,7 @@ def apply(self, ui):
                 return
             try:
-                with open(path) as f:
+                with codecs.open(path, encoding='utf-8') as f:
```
We use magic for making educated guesses about encoding, and we have a wrapper in  `alot.helpers.guess_encoding`.

something like this probably:
```python
with open(path, 'rb') as f:
     blob = f.read()
encoding = helpers.guess_encoding(blob)
self.envelope.parse_template(blob.decode(encoding))
```

dcbaker commented:
It would alos be nice if we had a test case for this.

three-comrades commented:
I used your suggestion and just added a debug logging statement for the encoding.

> It would alos be nice if we had a test case for this.

I added a test that makes sure that the template is not loaded as a byte string, which was the root cause of the crash.

dcbaker commented:
looks good to me, thanks for making those changes.

varac commented:
I can confirm that this work for me, great!
